src/Donations/Repositories/DonationRepository.php

```diff
@@ -183,9 +184,9 @@ public function insert(Donation $donation)
                     'post_date_gmt' => get_gmt_from_date($dateCreatedFormatted),
                     'post_modified' => $dateCreatedFormatted,
                     'post_modified_gmt' => get_gmt_from_date($dateCreatedFormatted),
-                    'post_status' => $donation->status->getValue(),
+                    'post_status' => $this->getPersistedDonationStatus($donation)->getValue(),
                     'post_type' => 'give_payment',
-                    'post_parent' => $donation->parentId ?? 0
+                    'post_parent' => $donation->type->isRenewal() ? give()->subscriptions->getInitialDonationId($donation->subscriptionId) : 0,
```

Comment:
Let's add a method to cleanup this duplicated logic for inserting and updating.

Reply:
Resolved in 81c61aff1